consensus/types/round_state.go

```diff
@@ -77,6 +79,32 @@ type RoundState struct {
 	LastValidators     *types.ValidatorSet `json:"last_validators"`
 }
 
+// Compressed version of the RoundState for use in RPC
+type RoundStateSimple struct {
+	HeightRoundStep   string          `json:"height/round/step"`
+	StartTime         time.Time       `json:"start_time"`
+	ProposalBlockHash cmn.HexBytes    `json:"proposal_block_hash"`
+	LockedBlockHash   cmn.HexBytes    `json:"locked_block_hash"`
+	ValidBlockHash    cmn.HexBytes    `json:"valid_block_hash"`
+	Votes             json.RawMessage `json:"height_vote_set"`
+}
+
+// Compress the RoundState to RoundStateSimple
+func (rs *RoundState) RoundStateSimple() RoundStateSimple {
+	votesJSON, err := rs.Votes.MarshalJSON()
+	if err != nil {
+		panic(err)
+	}
+	return RoundStateSimple{
```

Comment:
Can we inline this struct as it is the only occurence?

Reply:
I initially had it inlined but since it's involved in an RPC response I thought it would be better on its own so users could more easily find it